presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -368,66 +369,83 @@ protected Node visitShowPartitions(ShowPartitions showPartitions, Void context)
                     .orElseThrow(() -> new SemanticException(MISSING_TABLE, showPartitions, "Table '%s' does not exist", table));
 
-            List<TableLayoutResult> layouts = metadata.getLayouts(session, tableHandle, Constraint.alwaysTrue(), Optional.empty());
-            if (layouts.size() != 1) {
-                throw new SemanticException(NOT_SUPPORTED, showPartitions, "Table does not have exactly one layout: %s", table);
+            Optional<TableLayout> layout = getLayout(showPartitions, showPartitions.getTable(), showPartitions.getWhere());
+            if (!layout.isPresent()) {
+                return simpleQuery(
+                        selectList(identifier("no partitions")),
+                        aliased(
+                                values(row(new NullLiteral())),
+                                "x",
+                                ImmutableList.of("no partitions")));
             }
-            TableLayout layout = getOnlyElement(layouts).getLayout();
-            List<ColumnHandle> partitionColumns = layout.getDiscretePredicates()
-                    .orElseThrow(() -> new SemanticException(NOT_SUPPORTED, showPartitions, "Table does not have partition columns: %s", table))
-                    .getColumns();
-
-            /*
-                Generate a dynamic pivot to output one column per partition key.
-                For example, a table with two partition keys (ds, cluster_name)
-                would generate the following query:
-
-                SELECT
-                  partition_number
-                , max(CASE WHEN partition_key = 'ds' THEN partition_value END) ds
-                , max(CASE WHEN partition_key = 'cluster_name' THEN partition_value END) cluster_name
-                FROM ...
-                GROUP BY partition_number
-
-                The values are also cast to the type of the partition column.
-                The query is then wrapped to allow custom filtering and ordering.
-            */
-
-            ImmutableList.Builder<SelectItem> selectList = ImmutableList.builder();
-            ImmutableList.Builder<SelectItem> wrappedList = ImmutableList.builder();
-            selectList.add(unaliasedName("partition_number"));
-            for (ColumnHandle columnHandle : partitionColumns) {
-                ColumnMetadata column = metadata.getColumnMetadata(session, tableHandle, columnHandle);
-                Expression key = equal(identifier("partition_key"), new StringLiteral(column.getName()));
-                Expression value = caseWhen(key, identifier("partition_value"));
-                value = new Cast(value, column.getType().getTypeSignature().toString());
-                Expression function = functionCall("max", value);
-                selectList.add(new SingleColumn(function, new Identifier(column.getName())));
-                wrappedList.add(unaliasedName(column.getName()));
+
+            DiscretePredicates discretePredicates = layout.get().getDiscretePredicates()
```

Comment:
why not use `ConnectorTableLayout#getTablePartitioning` instead? There is method:
`getPartitioningColumns` there. I would expect it to be non-empty if `getTablePartitioning` is present (check state?)

There is a javadoc there: 
```
     * The partitioning of the table across the worker nodes.
     * <p>
     * If the table is node partitioned, the connector guarantees that each combination of values for
     * the distributed columns will be contained within a single worker.
```
so I am not exactly sure if it fits here.

Reply:
Unfortunately, it appears to be something completely different.